### ISSUE TITLE
say something does not terminate

### ISSUE THREAD

**Issue by ellenFan** [Uberi/speech_recognition#293]
after install the speech recognition, in examples directory, I tried python microphone_recognition.py Since I only installed python 2.7, I changed the first line of code to #!/usr/bin/env python
the program prints "say something" but it never terminates after that, here's what happened when I interrupted it with the keystroke

^CTraceback (most recent call last):
  File "microphone_recognition.py", line 11, in <module>
    audio = r.listen(source)
  File "/usr/lib/python2.7/site-packages/speech_recognition/__init__.py", line 649, in listen
    buffer = source.stream.read(source.CHUNK)
  File "/usr/lib/python2.7/site-packages/speech_recognition/__init__.py", line 163, in read
    return self.pyaudio_stream.read(size, exception_on_overflow=False)
  File "/usr/lib/python2.7/site-packages/PyAudio-0.2.11-py2.7-linux-i686.egg/pyaudio.py", line 608, in read
    return pa.read_stream(self._stream, num_frames, exception_on_overflow)
KeyboardInterrupt

I'm working on the intel edison board and the python version is python 2.7; pip version is pip 9.0.1; speech_recognition version is 3.7.1; pyaudio version is 0.2.11, pocket sphinx version 0.8, sphinxbase version 0.8

**Comment by Uberi** [comment]
Hi @ellenFan,

Does your Edison board have a sound card and microphone? If so, please post the output of the commands listed in the issue template - it's important information for diagnosing microphone issues.

**Comment by ellenFan** [comment]
Hi @Uberi 
I think I found the problem. I forgot to set the default value for ALSA (details could be found in https://bbs.archlinux.org/viewtopic.php?id=106726). Besides, the PocketSphinx I installed on the Edison is of version 0.8 and the PocketSphinx-python downloaded from command "pip install pocketsphinx", I think,  is of version 0.1.3. With PocketSphinx 0.8 installed, I also got an error in importing pocket sphinx but things worked out after 1) I set the default value for recording device and 2) flashed the Intel Edison and reinstall it with pocketsphinx 0.1.3. Thank you so much.
